core/domain/email_manager.py

```diff
@@ -1159,6 +1242,113 @@ def send_mail_to_notify_users_to_review(user_id, category):
+
+    Args:
+        reviewer_ids: list(str). A list of the Contributor Dashboard reviewer
+            user ids to notify.
+        reviewers_suggestion_email_infos:
+            list(list(ReviewableSuggestionEmailInfo)). A list of suggestion
+            email content info objects for each reviewer. These suggestion
+            email content info objects contain the key imformation about the
+            suggestions we're notifying reviewers about and will be used to
+            compose the email body for each reviewer.
+    """
+    email_subject = NOTIFY_CONTRIBUTOR_DASHBOARD_REVIEWERS_EMAIL_INFO[
+        'email_subject']
+    email_body_template = NOTIFY_CONTRIBUTOR_DASHBOARD_REVIEWERS_EMAIL_INFO[
+        'email_body_template']
+
+    if not feconf.CAN_SEND_EMAILS:
+        log_new_error('This app cannot send emails to users.')
+        return
+
+    if not config_domain.NOTIFY_CONTRIBUTOR_DASHBOARD_REVIEWERS_IS_ENABLED:
+        log_new_error(
+            'Notifying Contributor Dashboard reviewers must be enabled on the '
+            'config page in order to send reviewers the emails.'
+        )
+        return
+
+    if not reviewer_ids:
+        log_new_error('No Contributor Dashboard reviewers to notify.')
+        return
+
+    reviewer_user_settings = user_services.get_users_settings(reviewer_ids)
+    reviewer_usernames = [
+        reviewer_user_setting.username if reviewer_user_setting is not None else
+        None for reviewer_user_setting in reviewer_user_settings
+    ]
+    reviewer_emails = [
+        reviewer_user_setting.email if reviewer_user_setting is not None else
+        None for reviewer_user_setting in reviewer_user_settings
+    ]
+
+    send_email_infos = []
+    for index, reviewer_id in enumerate(reviewer_ids):
+        if not reviewers_suggestion_email_infos[index]:
+            log_new_error(
+                'There were no suggestions to recommend to the reviewer with '
+                'user id: %s.' % reviewer_id)
+            continue
+        elif not reviewer_emails[index]:
+            log_new_error(
+                'There was no email for the given reviewer id: %s.' % (
+                    reviewer_id))
+            continue
+        else:
+            list_of_suggestions_to_notify_reviewer_strings = []
+            for reviewer_suggestion_email_info in (
+                    reviewers_suggestion_email_infos[index]):
+                # Get the language of the suggestion.
+                language = utils.get_supported_audio_language_description(
+                    reviewer_suggestion_email_info.language_code)
+                # Set the language for question suggestions to be the empty
+                # string in order to use the a single suggestion template
+                # format.
+                if reviewer_suggestion_email_info.suggestion_type == (
+                        suggestion_models.SUGGESTION_TYPE_ADD_QUESTION):
+                    language = ''
```

Comment:
This seems a little hacky, and consequently a bit fragile since it requires all email info templates to have the same number of variables in the same order.

Could we instead use a table of lambdas that take in a dict of the values needed to populate each template? That allows the templates to be specialized while keeping the code here generic (e.g. don't populate ``dict['language']`` if there is no language to use).

Reply:
Yep, I think this is *much* better. Thanks!